source/docs/prismic/embed-fields.md

```diff
@@ -0,0 +1,231 @@
+
+In Front-Commerce we expose the embed fields in two ways;
+
+- Standalone Embed fields with an EmbedTransformer
+- Embed Fields in [`WysiwygV2`](/docs/advanced/theme/wysiwyg.html#lt-WysiwygV2-gt-usage) with [`PrismicWysiwyg`](/docs/advanced/theme/wysiwyg-platform.html#PrismicWysiwyg)
+
+In this section we will cover how to implement both of these methods, and how to implement your own custom embed fields.
+
+## Standalone Embed Fields
+
+You can learn how to [configure an embed field](https://prismic.io/docs/core-concepts/embed) in the prismic documentation.
+
+For these example let's say we want to create an album cover from an embed field.
+
+### Adding an Embed Field in your server-side
+
+First add an embed field in your schema using the [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type, for example:
+
+```graphql
+type MyAlbum {
+  title: String
+  artist: String
+  releaseDate: Date
+  cover: oEmbedContent
+}
+```
+
+Then you can implement the [`EmbedTransformer`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/loaders/transformers/Embed.js) in your loader which will parse the embed response from Prismic to a rich [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type.
+
+```js
+const AlbumLoader = (PrismicLoader) => {
+  const { EmbedTransformer } = PrismicLoader.transformers;
+
+  const contentTransformOptions = {
+    fieldTransformers: {
+      // ... other transformers
+      cover: new EmbedTransformer(),
+    },
+  };
+
+  return {
+    loadByUID: (uid) => {
+      return PrismicLoader.loadByUID("album", uid, contentTransformOptions);
+    },
+  };
+};
+```
+
+### Adding an Embed Field in your client-side
+
+We have added a `PrismicEmbed` component to handle the embed response in you client side.
+
+To simplify the usage between the component and the backend data we created a `PrismicEmbedFragment` which contains the minimum required data for the component, for example:
+
+```graphql
+#import "theme/modules/prismic/PrismicEmbed/PrismicEmbedFragment"
+
+fragment AlbumFragment on Album {
+  title
+  cover {
+    ...PrismicEmbedFragment
+  }
+}
+```
+
+The fragment will expose the `html`, `provider` and `type` fields in the [`oEmbedContent`](https://gitlab.com/front-commerce/front-commerce-prismic/-/blob/main/prismic/server/modules/prismic/core/schema.gql) type.
+
+```jsx
+import PrismicEmbed from "theme/modules/prismic/PrismicEmbed";
+
+const Album = (props) => {
+  return (
+    <div>
+      <h1>{props.title}</h1>
+      <PrismicEmbed
+        html={props.cover.html}
+        type={props.cover.type}
+        provider={props.cover.provider}
+      />
```

Comment:
**issue:** the example here highlights a possible responsibility issue. The `<Album />` component, shouldn't be aware of any of the `props.cover` field.\
It should only be the responsibility of the `<PrismicEmbed />` component.\
**Goal:** if we improve the `oEmbedContent` GraphQL interface, it should only impact the `PrismicEmbed` component & fragment. In this example, userland code will also be impacted.

**suggestion(easy):** we should only document props forwarding -> `<PrismicEmbed {...props.cover} />`

**suggestion(needs-code-refactoring):** or change the `<PrismicEmbed />` API to only accept one `prop` -> `<PrismicEmbed content={props.cover} />`

Reply:
Okey yea thanks I think I will go for the code refactoring, as it will align more with the Wysiwyg usage.

Reply:
added a refactoring [MR](https://gitlab.com/front-commerce/front-commerce-prismic/-/merge_requests/83) for this.